src/scripts/main.js

```diff
@@ -1 +1,83 @@
 'use strict';
+
+function createFirstPromise() {
+  return new Promise((resolve, reject) => {
+    let isClicked = false;
+
+    document.addEventListener('click', () => {
+      isClicked = true;
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      if (!isClicked) {
+        reject(new Error('First promise was rejected'));
+      }
+    }, 3000);
+  });
+}
+
+function createSecondPromise() {
+  return new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    });
+  });
+}
+
+function createThirdPromise() {
+  return new Promise((resolve, reject) => {
+    const isClicked = {
+      left: false,
+      right: false,
+    };
+
+    document.addEventListener('click', () => {
+      isClicked.left = true;
+      checkBothClicks();
+    });
```

Comment:
The 'click' event listener should be managed to prevent multiple resolutions of the promise. Consider removing the listener after the first resolution.